### PR TITLE
mgmt, local generation for Cosmos DB 2024-05-15 Stable

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#5205](https://github.com/Azure/sdk-release-request/issues/5205)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -495,6 +495,42 @@
...
+        {
+          "regex": true,
+          "code": "java\\.annotation\\.(attributeRemoved|attributeAdded)",
+          "old": ".*? com\\.azure\\.resourcemanager\\.cosmos\\.models.*",
+          "justification": "Jackson annotation changed."
+        },
+        {
+          "code": "java.method.removed",
+          "old": "method java.lang.Integer com.azure.resourcemanager.cosmos.models.ServiceResourceCreateUpdateParameters::instanceCount()",
+          "justification": "The property flatten disabled due to model changed to polymorphic."
```
Comment (XiaofeiCao) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:508]: We should mention these breaking changes in changelog, e.g.`ServiceResourceCreateUpdateProperties` is no longer flattened in `ServiceResourceCreateUpdateParameters`, as it's now polymorphic.

Comment (v-hongli1) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:508]: Added breaking changes to `CHANGELOG.md` in the new version.

File: sdk/resourcemanager/azure-resourcemanager-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -2,14 +2,23 @@
...
 
-### Bugs Fixed
+#### `models.ServiceResourceCreateUpdateParameters` was modified.
+
+* `instanceCount()` was removed.
+* `instanceSize()` was removed.
+* `serviceType()` was removed.
+* `withInstanceCount(java.lang.Integer)` was removed.
+* `withInstanceSize(com.azure.resourcemanager.cosmos.models.ServiceSize)` was removed.
+* `withServiceType(com.azure.resourcemanager.cosmos.models.ServiceType)` was removed.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-cosmos/CHANGELOG.md:14]: - Properties under `ServiceResourceCreateUpdateParameters` class is moved to `ServiceResourceCreateUpdateProperties` class of its "properties" property.

Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-cosmos/CHANGELOG.md:14]:
```suggestion
- Properties under `ServiceResourceCreateUpdateParameters` class is moved to `ServiceResourceCreateUpdateProperties` class of its "properties" property.
```

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**